### PR TITLE
accfg: Add SNAX gemm accelerator

### PR DESCRIPTION
**Pull request by JosseVanDelm**
Initial work on non-streamer gemm accelerator through accfg dialect

### CODE REVIEW

File: compiler/accelerators/snax_gemm.py [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+            return [
+                c_8 := arith.Constant.from_int_and_width(8, 32),
+                c_0xFF := arith.Constant.from_int_and_width(0xFF, 32),
+                subtraction_a_no_overflow := arith.AndI(subtraction_a, c_0xFF),
+                subtraction_b_no_overflow := arith.AndI(subtraction_b, c_0xFF),
+                shifted_subtraction_b := arith.ShLI(subtraction_b_no_overflow, c_8),
+                or_op := arith.OrI(subtraction_a_no_overflow, shifted_subtraction_b),
+            ], or_op.result
+
+        def _generate_size_config(Batch: int, M: int, K: int, N: int):
```
Comment (AntonLydike) [inline compiler/accelerators/snax_gemm.py:123]: nit: `Batch` should be lower case

Comment (AntonLydike) [reply compiler/accelerators/snax_gemm.py:123]: All of them should probably be lower case

File: compiler/accelerators/snax_gemm.py [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+            )
+            for ref in (a, b, c)
+        ]
+
+        return [
+            size_config,
+            *ptrs,
+            *_get_constants(constants, 32),
+            _generate_subtract_config(zpa, zpb)
+            # ([barrier_enable:=arith.Constant.from_int_and_width(1, 5)],barrier_enable.result) #Always enable barrier
```
Comment (AntonLydike) [inline compiler/accelerators/snax_gemm.py:184]: Commented code

**Review by AntonLydike — APPROVED** [review]
:heart:

File: compiler/accelerators/snax_gemm.py (hunk 2 above)
Comment (JosseVanDelm) [reply compiler/accelerators/snax_gemm.py]: removed!

File: compiler/accelerators/snax_gemm.py (hunk 1 above)
Comment (JosseVanDelm) [reply compiler/accelerators/snax_gemm.py]: Okay!

**Pull request merged**